func var declared before func type

The following program `sample.go` triggers a panic:

```go
package main
  
var f F

type F func(int)

func main() {
    println("ok")
}
```

Expected result:
```console
$ go run ./sample.go
ok
```

Got:
```console
$ yaegi ./sample.go
./sample.go:3:7: incomplete type 

goroutine 1 [running]:
github.com/containous/yaegi/interp.(*itype).frameType(0xc000288410, 0x7400000000000010, 0xc0001a8050)
        /home/marc/go/src/github.com/containous/yaegi/interp/type.go:965 +0x1c2
github.com/containous/yaegi/interp.(*scope).add(0xc00017a360, 0xc000288410, 0xc000298500)
        /home/marc/go/src/github.com/containous/yaegi/interp/scope.go:156 +0x49
github.com/containous/yaegi/interp.(*Interpreter).gta.func1(0xc000298300, 0xc00017dce0)
        /home/marc/go/src/github.com/containous/yaegi/interp/gta.go:85 +0x1c73
github.com/containous/yaegi/interp.(*node).Walk(0xc000298300, 0xc0001d7d18, 0x0)
        /home/marc/go/src/github.com/containous/yaegi/interp/interp.go:164 +0xc9
github.com/containous/yaegi/interp.(*node).Walk(0xc000298200, 0xc0001d7d18, 0x0)
        /home/marc/go/src/github.com/containous/yaegi/interp/interp.go:168 +0x70
github.com/containous/yaegi/interp.(*node).Walk(0xc000298000, 0xc0001d7d18, 0x0)
        /home/marc/go/src/github.com/containous/yaegi/interp/interp.go:168 +0x70
github.com/containous/yaegi/interp.(*Interpreter).gta(0xc00027c6e0, 0xc000298000, 0xc000182060, 0x4, 0x7fffc6eb0701, 0x1e, 0x4, 0xc000298000, 0x0, 0x0, ...)
        /home/marc/go/src/github.com/containous/yaegi/interp/gta.go:17 +0x131
github.com/containous/yaegi/interp.(*Interpreter).Eval(0xc00027c6e0, 0xc000290000, 0x48, 0x248, 0xc000290000, 0x48, 0x0, 0x1)
        /home/marc/go/src/github.com/containous/yaegi/interp/interp.go:311 +0x117
main.main()
        /home/marc/go/src/github.com/containous/yaegi/cmd/yaegi/yaegi.go:82 +0x42e
```

--- a/_test/fun11.go
+++ b/_test/fun11.go
@@ -1,0 +1,12 @@
+package main
+
+var f F
+
+type F func(int)
+
+func main() {
+	println("ok")
+}
+
+// Output:
+// ok

--- a/interp/gta.go
+++ b/interp/gta.go
@@ -52,7 +52,7 @@ func (interp *Interpreter) gta(root *node, rpath, pkgID string) ([]*node, error)
 					val = src.rval
 				}
 				if typ.incomplete {
-					// Come back when type is known
+					// Come back when type is known.
 					revisit = append(revisit, n)
 					return false
 				}
@@ -78,6 +78,11 @@ func (interp *Interpreter) gta(root *node, rpath, pkgID string) ([]*node, error)
 			l := len(n.child) - 1
 			if n.typ = n.child[l].typ; n.typ == nil {
 				if n.typ, err = nodeType(interp, sc, n.child[l]); err != nil {
+					return false
+				}
+				if n.typ.incomplete {
+					// Come back when type is known.
+					revisit = append(revisit, n)
 					return false
 				}
 			}

